src/com/amazon/ion/impl/_Private_IonReaderBuilder.java

```diff
@@ -109,54 +148,103 @@ public IonReader build(byte[] ionData, int offset, int length)
      * @return true if the first 'length' bytes in 'buffer' match the first 'length' bytes in the binary IVM.
      */
     private static boolean startsWithIvm(byte[] buffer, int length) {
-        for (int i = 0; i < length; i++) {
-            if (_Private_IonConstants.BINARY_VERSION_MARKER_1_0[i] != buffer[i]) {
-                return false;
-            }
+        if (length >= _Private_IonConstants.BINARY_VERSION_MARKER_SIZE) {
+            return buffer[0] == (byte) 0xE0
+                && buffer[3] == (byte) 0xEA;
```

Comment:
shouldn't this check the 2nd and 3rd bytes too?

Reply:
The first and fourth bytes are enough to determine that this is *some* binary Ion IVM. It's up to the cursor to determine the version (denoted by the middle two bytes), and to branch its parsing logic accordingly.